### PR TITLE
Fix product integer overflow

### PR DESCRIPTION
**Pull request by JoOkuma**
This `np.prod` call was overflowing when converting our data and resulting in invalid chunks.

### CODE REVIEW

**Review by ziw-liu — COMMENTED** [review]
Thanks for spotting this!

Out of curiosity, what's the shape of the input shape here? `int64` should be enough for 1 billion $2048^3$ volumes.

**Comment by JoOkuma** [comment]
Some DaXi volumes were approximately 3500x2048x2048 🥲

**Comment by ziw-liu** [comment]
> Some DaXi volumes were approximately 3500x2048x2048 🥲

But this should still be well within the range of `int64`. Was the shape in `uint16` somehow?

File: iohub/convert.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def _gen_chunks(self, input_chunks):
         # it's OK if a single image is larger than MAX_CHUNK_SIZE
         while (
             chunks[-3] > 1
-            and np.prod(chunks) * bytes_per_pixel > MAX_CHUNK_SIZE
+            and np.prod(chunks, dtype=float) * bytes_per_pixel > MAX_CHUNK_SIZE
```
Comment (ziw-liu) [inline iohub/convert.py:360]: For example I would expect this to be sufficient for any array that can fit on our data partitions:

```suggestion
            and np.prod(chunks, dtype=np.int64) * bytes_per_pixel > MAX_CHUNK_SIZE
```

**Review by ziw-liu — APPROVED** [review]

**Pull request merged**